Drop `DIST_MODULE_PATH`

dochang opened the pull request:
People may install python into a non-standard location other than `/usr`, such as `$HOME`.  That means `DIST_MODULE_PATH` should be always based on `sys.prefix`.

spencergibb commented on lib/battleschool/constants.py:
```diff
@@ -11,12 +11,7 @@ def get_config(env_var, default):
-    DIST_MODULE_PATH = os.path.join(sys.prefix, 'share/battleschool/')
-else:
-    DIST_MODULE_PATH = '/usr/share/battleschool/'
+DIST_MODULE_PATH = os.path.join(sys.prefix, 'share/battleschool/')
```
To fix #46 I had to make sure that mac users _are not_ installing to `sys.prefix` (which for me is `/System/Library/Frameworks/Python.framework/Versions/2.7`).

How do we accommodate both?

dochang committed:
Drop `DIST_MODULE_PATH`

Install data files into the package directory instead of `sys.prefix`.
So we can find them without `DIST_MODULE_PATH`.  This closes all related
issues.

dochang commented:
Now `DIST_MODULE_PATH` has been removed. Data files location will not be related with `sys.prefix`.

spencergibb commented:
You'll have to walk me through all of your changes. I don't understand them. Also, please rebase.

dochang commented:
This patch installs data files (the `share` directory) into the same location of `*.py` files. No files will be installed into `/usr/local/share/battleschool` or `os.path.join(sys.prefix, 'share/battleschool')` any more.

The effect is: we can access `library/mac_pkg` or `defaults/hosts` via `pkg_resources` API, rather than `DIST_MODULE_PATH`. So, we can remove `DIST_MODULE_PATH` completely.

dochang commented:
This patch has been rebased to master. Should I rebase it back to the original commit (49d8c12) ?

spencergibb commented:
Installs on my mac to `/Library/Python/2.7/site-packages/battleschool` which according to the [About System Integrity Protection](https://support.apple.com/en-us/HT204899) page is ok.

spencergibb commented on lib/battleschool/constants.py:
```diff
@@ -1,5 +1,6 @@
 import os
 import sys
+from pkg_resources import resource_filename
```
I don't like requiring `setuptools` for the runtime. Is there a better way to do this?

spencergibb commented:
The following works for me:

```
def get_share_path(*paths):
    lib_path = os.path.dirname(__file__)
    share_path = os.path.join(lib_path, 'share')
    return os.path.join(share_path, *paths)
```

@dochang any objections?

dochang commented:
Dropped `pkg_resources` and `setuptools`.  @spencergibb
